File: api/src/main/java/org/ftt/familytasktracking/tasks/scheduler/SchedulerMode.java
package org.ftt.familytasktracking.tasks.scheduler;

/**
 * Enum for the Scheduler Mode
 */
public enum SchedulerMode {
    /**
     * Cron uses a Cron Expression for resolving Execution Dates
     */
    CRON {
        @Override
        public TaskScheduler createScheduler() {
            return new CronTaskScheduler();
        }
    },
    /**
     * Milli Interval uses an actual Milli Interval for resolving Execution Dates
     */
    MILLI_INTERVAL {
        @Override
        public TaskScheduler createScheduler() {
            return new MilliIntervalTaskScheduler();
        }
    },
    /**
     * Deactivates the Scheduler
     */
    DEACTIVATED {
        @Override
        public TaskScheduler createScheduler() {
            return null;
        }
    };

    public abstract TaskScheduler createScheduler();
}

File: frontend/types/scheduler-mode.ts
export enum SchedulerMode {
  /**
   * Cron uses a Cron Expression for resolving Execution Dates
   */
  Cron = "CRON",
  /**
   * Milli Interval uses an actual Milli Interval for resolving Execution Dates
   */
  MilliInterval = "MILLI_INTERVAL",
  /**
   * Deactivates the Scheduler
   */
  Deactivated = "DEACTIVATED",
}

export function getTranslatedSchedulerModeValue(mode: SchedulerMode) {
  switch (mode) {
    case SchedulerMode.Cron:
      return "Cron";
    case SchedulerMode.MilliInterval:
      return "Interval";
    case SchedulerMode.Deactivated:
      return "Inactive";
    default:
      return "Invalid";
  }
}
